### PR TITLE
boards/stm32: add support to STM32F401RC-RS485 board

### PR DESCRIPTION
**Pull request by rcsim**
## Summary
The STM32F401RC-RS485 is a board with RS485 support, SD Card, buttons, leds, temperature sensor, adc and etc.

## Impact
User will be able to use this new board STM32F401RC-RS485.

## Testing
Build basic shell application and ran on STM32F401RC-RS485 board.

P.S.: Documentation will be sent in a different commit ad we need to format it.

### CODE REVIEW

File: boards/Kconfig [hunk 1]
```diff
@@ -1397,6 +1397,8 @@ config ARCH_BOARD_BEAGLEBONE_BLACK
 	---help---
 		This is the port of NuttX to the TI Beaglebone Black board.
 
+
+
```
Comment (xiaoxiang781216) [inline boards/Kconfig:1401]: why change

File: boards/Kconfig [hunk 2]
```diff
@@ -1432,14 +1434,6 @@ config ARCH_BOARD_PIC32MZ_STARTERKIT
 
 		See www.microchip.com for further information.
 
-config ARCH_BOARD_NUCLEO_F429ZI
```
Comment (xiaoxiang781216) [inline boards/Kconfig:1435]: why delete

Comment (rcsim) [reply boards/Kconfig]: Thanks for your feedback. I will rework that.

File: boards/Kconfig [hunk 3]
```diff
@@ -1589,6 +1589,17 @@ config ARCH_BOARD_NUCLEO_F401RE
...
+	bool "STM32F401RC-RS485 Board"
+	depends on ARCH_CHIP_STM32F401RC
+	select ARCH_HAVE_LEDS
+	select ARCH_HAVE_BUTTONS
+	select ARCH_HAVE_IRQBUTTONS
+	---help---
+		This is a minimal configuration that supports low-level test of the
+		STM32F401RC-RS485 in the NuttX source tree.
+
+
```
Comment (acassis) [inline boards/Kconfig:1602]: Please remove a extra empty line

Comment (rcsim) [reply boards/Kconfig]: Hi Alan,
Extra line has been removed!

Thanks!

**Review by rcsim — COMMENTED** [review]
Thanks for your review!
Fixes has been done as requested.

Best regards,

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**